cinema/views.py

```diff
@@ -4,135 +1,39 @@
 from rest_framework import viewsets
-from rest_framework.pagination import PageNumberPagination
-
+from cinema.serializers import MovieSerializer
+from rest_framework.permissions import IsAuthenticated
+from rest_framework.viewsets import ModelViewSet
+from user.permissions import IsAdminOrIfAuthenticatedReadOnly
 from cinema.models import Genre, Actor, CinemaHall, Movie, MovieSession, Order
 
-from cinema.serializers import (
-    GenreSerializer,
-    ActorSerializer,
-    CinemaHallSerializer,
-    MovieSerializer,
-    MovieSessionSerializer,
-    MovieSessionListSerializer,
-    MovieDetailSerializer,
-    MovieSessionDetailSerializer,
-    MovieListSerializer,
-    OrderSerializer,
-    OrderListSerializer,
-)
 
-
-class GenreViewSet(viewsets.ModelViewSet):
+class GenreViewSet(ModelViewSet):
     queryset = Genre.objects.all()
-    serializer_class = GenreSerializer
+    permission_classes = [IsAdminOrIfAuthenticatedReadOnly]
 
 
-class ActorViewSet(viewsets.ModelViewSet):
-    queryset = Actor.objects.all()
-    serializer_class = ActorSerializer
+class CinemaHallViewSet(ModelViewSet):
+    queryset = CinemaHall.objects.all()
+    permission_classes = [IsAdminOrIfAuthenticatedReadOnly]
 
 
-class CinemaHallViewSet(viewsets.ModelViewSet):
-    queryset = CinemaHall.objects.all()
-    serializer_class = CinemaHallSerializer
+class ActorViewSet(ModelViewSet):
+    queryset = Actor.objects.all()
+    permission_classes = [IsAdminOrIfAuthenticatedReadOnly]
 
 
 class MovieViewSet(viewsets.ModelViewSet):
```

Comment:
There is an inconsistency in the import statement for `ModelViewSet`. In this line, `viewsets.ModelViewSet` is used, but in other classes, `ModelViewSet` is used directly. Consider using a consistent import style for clarity.